tests/acceptance/test_update_control.py

```diff
@@ -443,19 +453,125 @@ def make_and_deploy_artifact():
+            "case": {
+                "pause_map": {
+                    "id": MUID,
+                    "states": {"ArtifactCommit_Enter": {"action": "pause",},},
+                },
+                "pause_state": "ArtifactVerifyReboot",
+                "continue_map": {
+                    "id": MUID,
+                    "states": {
+                        "ArtifactInstall_Enter": {"action": "fail",},
+                        "ArtifactCommit_Enter": {"action": "continue",},
+                    },
+                },
+            },
+        },
+        {
+            "name": "Cleanup after failure",
+            "case": {
+                "pause_map": {
+                    "id": MUID,
+                    "states": {"ArtifactCommit_Enter": {"action": "pause",},},
+                },
+                "pause_state": "ArtifactVerifyReboot",
+                "continue_map": {
+                    "id": MUID,
+                    "states": {
+                        "ArtifactInstall_Enter": {"action": "fail",},
+                        "ArtifactCommit_Enter": {"action": "fail",},
+                    },
+                },
+            },
+        },
+    ]
+
+    @pytest.mark.min_mender_version("2.7.0")
+    @pytest.mark.parametrize(
+        "case_name,case",
+        [
+            (case["name"], case["case"])
+            for case in test_update_control_maps_cleanup_cases
+        ],
+    )
+    def test_update_control_maps_cleanup(
+        self,
+        case_name,
+        case,
+        setup_board,
+        connection,
+        second_connection,
+        setup_mock_server,
+        bitbake_variables,
+        bitbake_path,
+    ):
+        def wait_for_state(state_to_wait_for):
+            log = []
+            attempts = 10
+            while attempts > 0:
+                output = connection.run(
+                    "cat /data/logger-update-module.log 2>/dev/null || true"
+                ).stdout.strip()
+                log = [line.split()[1] for line in output.split("\n") if len(line) > 0]
+
+                if state_to_wait_for in log:
+                    break
+
+                time.sleep(2)
+
+                attempts -= 1
+            else:
+                pytest.fail(f"Could not find {state_to_wait_for} in log")
+            return log
+
+        try:
+            start_and_ready_mender_client(connection, second_connection)
+
+            # First deployment sends the "pause" control map via Server API with the
+            # update; then once the client is paused, the map is overridden via DBus API
+            make_and_deploy_artifact(
+                connection,
+                bitbake_variables["MENDER_DEVICE_TYPE"],
+                deployment_id=MUID,
+                update_control_map=case["pause_map"],
+            )
+            wait_for_state(case["pause_state"])
+            set_update_control_map(connection, case["continue_map"])
+            wait_for_state("Cleanup")
```

Comment:
As a safeguard to make sure the map is actually working (and not just being silently ignored or simply not picked up), maybe check the status of the first deployment here, either success or failure depending on the test case?

Reply:
Okey, fair enough.